### ISSUE TITLE
tower link placement mistakes

### ISSUE THREAD

**Issue by sparr** [TooAngel/screeps#445]
https://i.imgur.com/6qfdLZZ.png
https://i.imgur.com/xYLx3og.png

two example so far. I'll try to debug the code when I get a base far enough to build links.

**Comment by sparr** [comment]
![](https://i.imgur.com/TDtRSKc.png)

**Comment by TooAngel** [comment]
I guess you got the idea.

Anyway will explain the basic idea :-):
Having `link`s close to the entrances, so that carries from external rooms can drop energy into the links. Next to the link have a position for a `towerfiller` and next to that one the `tower`. `link` and `tower` should be adjacent to the path so that carries drop energy in both.
When under attack a towerfiller should be spawn for each towerfiller position, transferring energy between link and tower.
The `filler` creep at the storage starts to transfer energy from the `storage` to the `link` at that link starts to send energy to others.

**Comment by sparr** [comment]
@TooAngel I understand the idea. The problem illustrated here is that the towerfiller (yellow "t") is being placed not-adjacent to the link (blue "Li") or the tower (blue "To") sometimes.

**Comment by TooAngel** [comment]
Yes, agree, got it. Needs to be fixed :-) Good to have the issue ...

If you or someone else will look into it at some point, I just want to reference the base idea how it should have looked it. Fixing would be cool, improving the idea - even better ;-)

**Comment by sparr** [comment]
Is it important to have one link and filler per tower? It seems like most layouts have two of the towers close together. Combining them to use one link and two fillers seems trivial. Using one link and one filler would be a little trickier, but not much.

**Comment by TooAngel** [comment]
True, we could also do more tower.

Currently it iterates over all exit paths and places the combination on each. If there are less exits, we could extend the combination by more towers.

**Comment by sparr** [comment]
https://github.com/TooAngel/screeps/blob/master/src/prototype_room_init.js#L192-L204

Problem is here. I'm going to re-work this.
